overlord/devicestate/firstboot.go

```diff
@@ -34,13 +35,21 @@ import (
 	"github.com/snapcore/snapd/overlord/assertstate"
 	"github.com/snapcore/snapd/overlord/auth"
 	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/snapstate/backend"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/release"
 	"github.com/snapcore/snapd/snap"
 )
 
 var errNothingToDo = errors.New("nothing to do")
 
+var snapInfoFromFile = snapInfoFromFileImpl
+
+func snapInfoFromFileImpl(snapPath string) (*snap.Info, error) {
+	info, _, err := backend.OpenSnapFile(snapPath, nil)
```

Comment:
InstallPath already does that too and after having checked the hash,  still would prefer an approach that leverages that

Reply:
Sure, I updated the code, requires a bit more (code) churn but avoids the double reading.

Reply:
*If* this looks reasonable I can split the InstallPath change out and push it as a septate PR without any new functionality. This way its probably easier to review.